src/main/java/org/jabref/logic/search/SearchQuery.java

```diff
@@ -59,6 +59,19 @@ public String toString() {
         return query;
     }
 
+    @Override
+    public boolean equals(Object other) {
+        if (other instanceof SearchQuery) {
```

Comment:
```suggestion
        if (other instanceof SearchQuery searchQuery) {
        return other.query...
```
Use the new java pattern matching :)

Reply:
I remembered there was something like that, but was too lazy to look it up 😅 thanks.